caproto/asyncio/utils.py

```diff
@@ -178,6 +236,28 @@ def submit(self, callback, *args, **kwargs):
         self.callbacks.put((callback, args, kwargs))
 
 
+async def _create_bound_tcp_socket(addr, port):
+    """Create a TCP socket and bind it to (addr, port)."""
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM, 0)
+    sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    sock.setblocking(False)
+    sock.bind((addr, port))
+    return sock
+
+
+def _create_udp_socket():
+    """Create a UDP socket for usage with a datagram endpoint."""
+    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
+    # Python says this is unsafe, but we need it to have
```

Comment:
I recall the context but it might be helpful for posterity if there were a link in this comment.

Reply:
Comment from @tacaswell as I recall; not sure of the link.